portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,16 +38,10 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> comments = getDatastoreComments();
-
-    String json = "{\"comments\": [";
-    for (String comment : comments)
-      json = json.concat("{\"value\": \"" +  comment + "\" },");
-    //replaces last comma with JSON enclosing bracket
-    json = "".concat(json.substring(0, json.length() - 1) + "]}");
+    Gson gson = new Gson();
     
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(gson.toJson(getDatastoreComments()));
```

Comment:
This is a perfect example of library code saving tons of lines! Sometimes spending a bit of extra time looking into what libraries are out there can help prevent you from "reinventing the wheel".

[Guava](https://github.com/google/guava) contains a bunch of helpful open source libraries from Google. A ton of Java code I've written relies on the data structures provided in `com.google.common.collect`.

Reply:
I'll definitely have to check that out! When it comes to the capstone I really want to work on making clean code, especially since I won't have these tutorials as a guide.